### PR TITLE
Adds support for excluding some search actions

### LINKED ISSUE #76: Add support for the analytics boolean
Building on the topic in the community at: https://discourse.algolia.com/t/exclude-logged-in-agents-from-zendesk-search-results/6047

The request here is to add support for `analytics` here:

```
algoliasearchZendeskHC({ 
     applicationId: 'APP_ID', 
     apiKey: 'API_KEY', 
     subdomain:  'SUBDOMAIN', 
     analytics: false
   }); 
```

It is suggested that this work build on https://github.com/algolia/algoliasearch-zendesk/pull/74

### PR DESCRIPTION
**Pull request by YesThatAllen**
Based on the queries I see being sent over, this is where I think the `analytics` key needs to be set.

Closes #76

### CODE REVIEW

File: app/src/instantsearch.js [hunk 1]
```diff
@@ -190,6 +191,7 @@ class InstantSearch {
 
     this.instantsearch.addWidget(
       instantsearch.widgets.hits({
+        analytics,
```
Comment (Jerska) [inline app/src/instantsearch.js]: I don't believe that the hits widget has a `hits` parameter.

**Comment by YesThatAllen** [comment]
Glad to help.. feel free to squash these commits at merge.

File: app/src/instantsearch.js [hunk 2]
```diff
@@ -42,6 +42,7 @@ class InstantSearch {
         }
       },
       searchParameters: {
+        analytics,
```
Comment (Jerska) [inline app/src/instantsearch.js]: I might be reading that wrong, but I believe you're not getting `analytics` in the function parameters (but are doing so in the `render` function).
You'll need to remove it from `render` and put it in this `constructor`.

File: app/src/autocomplete.js [hunk 3]
```diff
@@ -76,6 +76,7 @@ class Autocomplete {
       const sizeModifier = this._sizeModifier(dropdownMenuWidth);
       const nbSnippetWords = this._nbSnippetWords(dropdownMenuWidth);
       const params = {
+        analytics,
```
Comment (Jerska) [inline app/src/autocomplete.js]: Same here, but the other way around: you're not getting `analytics` in the `render` function parameters, but in the `constructor`. You'll need to swap this.

**Comment by Jerska** [comment]
This will still not work. Here's you're just passing the parameters to our wrappers, but you need to forward it to the actual algolia libraries calls.

That's both places described in my first comment.

---

In `autocomplete.js`, you'll need to:
1. Remove the unnecessary `analytics` from the `constructor` parameters.
2. Add `analytics` to the `render` function parameters.
3. Modify, in `render` the params, like so

   ```diff
     const params = {
   +   analytics, 
       hitsPerPage, 
       facetFilters: `["locale.locale:${locale}"]`, 
       highlightPreTag: '<span class="aa-article-hit--highlight">', 
       highlightPostTag: '</span>', 
       attributesToSnippet: [`body_safe:${nbSnippetWords}`], 
       snippetEllipsisText: '...' 
    };
   ```

---

In `instantsearch.js`, you'll need to:
1. Remove the unnecessary `analytics` from the `render` parameters.
2. Modify, in `contructor` the params, like so

   ```diff
         searchParameters: {
           attributesToSnippet: ['body_safe:40'],
           highlightPreTag: '<span class="ais-highlight">',
           highlightPostTag: '</span>',
           snippetEllipsisText: '...'
         },
   ```

**Comment by Jerska** [comment]
How are your tests of the library doing on your Help Center?

**Comment by YesThatAllen** [comment]
> How are your tests of the library doing on your Help Center?

Poorly. Per the local testing docs, `yarn` installs, but I get only

```
$ PORT=3005 yarn dev
yarn run v1.12.1
error Command "dev" not found.
```

Thoughts? but honestly, relying on me to test this may not be the best plan long-term..  I'm far out of my league on this.

**Comment by Jerska** [comment]
> Thoughts? but honestly, relying on me to test this may not be the best plan long-term.. I'm far out of my league on this.

No worries! I'm trying to point you in the right directions as you've started implementing it, but I can definitely take back ownership on it if this is what you want.

**Comment by YesThatAllen** [comment]
Fantastic. Very happy for you to take this & run with it. Looking forward to the feature add!

**Pull request closed without merging**